src/advanced/src/stores/productStore.ts

```diff
@@ -0,0 +1,44 @@
+import { create } from 'zustand';
+
+import { products } from '../data/products';
+
+import type { ProductModel } from '../types/cart';
+
+type State = {
+  products: ProductModel[];
+  lastAddedProduct: ProductModel | null;
+};
+
+type Action = {
+  updateProductQuantity: (targetProduct: ProductModel, newQuantity: number) => void;
+  updateProductPrice: (targetProduct: ProductModel, newPrice: number) => void;
+  updateLastAddedProduct: (targetProduct: ProductModel) => void;
+};
+
+export const useProductStore = create<State & Action>((set) => ({
+  products: products,
+  lastAddedProduct: null,
+
+  updateProductQuantity: (targetProduct, newQuantity) =>
+    set((state) => ({
+      products: state.products.map((product) => {
+        if (product.id === targetProduct.id) {
+          return { ...product, quantity: newQuantity };
+        } else {
+          return product;
+        }
+      }),
+    })),
+  updateProductPrice: (targetProduct, newPrice) => {
+    set((state) => ({
+      products: state.products.map((product) => {
+        if (product.id === targetProduct.id) {
+          return { ...product, price: newPrice };
+        } else {
+          return product;
+        }
+      }),
+    }));
+  },
```

Comment:
이 두 개를 공통으로 묶는 방법도 좋을 거 같아요!
이렇게 되면 수량이나 가격 외에 다른 값에도 대응하기 유용할 것 같습니다!

```suggestion
  updateProduct: (targetProduct, updated: Partial<ProductModel>) =>
    set((state) => ({
      products: state.products.map((product) => {
        if (product.id === targetProduct.id) {
          return { ...product, ...updated };
        } else {
          return product;
        }
      }),
    })),

```

Reply:
오 안 그래도 이 부분 고민이 많았는데요, 어떤 걸 업데이트 할 것인지 명확하게 하고 싶어서 분리해두었습니다!
업데이트 할 항목이 늘어났을 때 말씀주신 방법이 불필요한 중복을 줄여 깔끔해질 것 같네요 :)
제안 감사합니다 👍👍